stdlib/test/builtin/test_bool.mojo

```diff
@@ -145,6 +145,11 @@ def test_comparisons():
     assert_true(True <= True)
 
 
+def test_float_conversion():
+    assert_equal(float(True), 1.0)
+    assert_equal(float(False), 0.0)
```

Comment:
please test `__float__` directly to avoid accidental conversion for these primitive types